### PR TITLE
First Version to Git

### PR DESCRIPTION
**Pull request by stevekille**
This is a proposed XEP,   I'll send an email about it

Steve

### CODE REVIEW

File: inbox/optimisedS2S.xml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+<section1 topic='Glossary' anchor='glossary'>
+  <p>OPTIONAL.
+  </p>
+  <p> Sections without text to be removed prior to sending doc for XSF review.</p>
+</section1>
+<section1 topic='Use Cases' anchor='usecases'>
+  <p>STRONGLY RECOMMENDED.</p>
+</section1>
+<section1 topic='Business Rules' anchor='rules'>
+  <p>Typically a pair of XMPP servers connecting using this protocl will communcicate with multiple domains (e.g., a base domain and a MUC domain).   It is generally desirable to configure things so that all communications will share the same link, rather than establishing separate links for each domain.  Two or more connections MAY be initiated from one server to the other but this is NOT RECOMMENDED.   </p>
```
Comment (ralphm) [inline inbox/optimisedS2S.xml]: Spelling

File: inbox/optimisedS2S.xml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  <p>
+    An XMPP server receiving messages over such a link should appropriately validate to and from elements of inbound messages.   The rules for this SHOULD be controlled by an priori agreement.  An inbound connection will generally be associated with several peer domains.   A RECOMMENDED approach is to consider each of these peers in turn and validate in the manner of a peer XMPP server connected using RFC 6020 for that domain.  In the event that an inbound message is not considered to be valid, it should be handled in a manner that this invalid message would be handled if it arrived over standard S2S.
```
Comment (ralphm) [inline inbox/optimisedS2S.xml]: I assume this wants to refer to RFC 6**1**20.

File: inbox/optimisedS2S.xml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  </p>
+</section1>
+<section1 topic='Implementation Notes' anchor='impl'>
+  <section2 topic="General">
+    <p>
+      In simple terms, this can be considered as operation of RFC 6121 communication between a pair of XMPP servers without the preliminary negotiation done in RFC 6120.    It might be considered that the start point is the
+      DONE box in Figure 3 of RFC 6121.  The TCP connection is opened and messages start to flow.  All configuration informaition, including choice of port is handled by a the a priori configuration.</p>
```
Comment (ralphm) [inline inbox/optimisedS2S.xml]: That figure is in RFC 612**0**

Comment (stevekille) [inline inbox/optimisedS2S.xml]: Ralph,

 

Thanks!   Fixed this and the other error you pointed out.   Also ran spell check (thought I’d done this) which found a couple of errors and I hope is now dealing with your spell check comment.

 

Change pushed to GitHub, and I now need to ask Kev about the next Git step!

 

 

Steve

 

> +      DONE box in Figure 3 of RFC 6121.  The TCP connection is opened and messages start to flow.  All configuration informaition, including choice of port is handled by a the a priori configuration.</p>

That figure is in RFC 6120

**Comment by linuxwolf** [comment]
The preference is to use all lowercase names for XEPs, to avoid problems with case-insensitive filesystems and tools.

We will rename it to optimized-s2s.xml, as well as merge all changes into a single commit.

**Comment by stevekille** [comment]
Matt,

 

Thanks.   I’ll change my filename too.   

 

 

Steve

**Comment by linuxwolf** [comment]
Merged as 79b3628e88.

Thanks for the submission!

**Pull request closed without merging**